File: cinemaProject/src/main/java/com/example/cinemaProject/service/BillService.java
package com.example.cinemaProject.service;

import org.springframework.http.ResponseEntity;

import java.io.IOException;
import java.util.Map;

public interface BillService {
    public String generateReport(String nume,String prenume,String email,String adresa,String
            city,String state) throws IOException;

}

File: src/app/service/BillService.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";


@Injectable({
  providedIn: 'root'
})
export class BillService {

  baseURL: string = "http://localhost:8081/bill";

  constructor(private httpClient: HttpClient) {
  }

  createBill(nume: any, prenume:any,email:any,adresa:any,
             city:any,state:any) {
    const url = `${this.baseURL}/createBill?nume=${nume}&prenume=${prenume}&email=${email}&adresa=${adresa}&city=${city}
    &state=${state}`;
    return this.httpClient.put(url,{});
  }

}

